commons/all/src/main/java/org/infinispan/commons/jdkspecific/ClasspathURLStreamHandler.java

```diff
@@ -0,0 +1,47 @@
+package org.infinispan.commons.jdkspecific;
+
+import java.io.IOException;
+import java.net.URL;
+import java.net.URLConnection;
+import java.net.URLStreamHandler;
+import java.net.URLStreamHandlerFactory;
+import java.security.AccessController;
+import java.security.PrivilegedAction;
+
+public class ClasspathURLStreamHandler implements URLStreamHandlerFactory {
+
+   private static final URLStreamHandlerFactory INSTANCE = new ClasspathURLStreamHandler();
+
+   public static void register() {
+      if (System.getSecurityManager() != null) {
+         AccessController.doPrivileged((PrivilegedAction<Void>) () -> {
+            URL.setURLStreamHandlerFactory(INSTANCE);
+            return null;
+         });
+      } else {
+         URL.setURLStreamHandlerFactory(INSTANCE);
+      }
+   }
+
+   @Override
+   public URLStreamHandler createURLStreamHandler(String protocol) {
+      if ("classpath".equals(protocol)) {
+         return new URLStreamHandler() {
+            @Override
+            protected URLConnection openConnection(URL u) throws IOException {
+               String path = u.getPath();
+               URL resource = ClassLoader.getSystemClassLoader().getResource(path);
+               if (resource == null) {
+                  resource = Thread.currentThread().getContextClassLoader().getResource(path);
```

Comment:
The context class loader may be `null`. 

IMO it would be better to try the context class loader first (if non-null) and fall back to the system class loader.